File: PackageConfig/PackageConfig/ViewController.m
//
//  ViewController.m
//  PackageConfig
//
//  Created by point on 2017/5/10.
//  Copyright © 2017年 dacai. All rights reserved.
//

#import "ViewController.h"
#import "DCTool.h"
#import "DCConfig.h"

@interface ViewController ()

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    DCTool * t1 = [[DCTool alloc]init];
    DCTool * t2 = [[DCTool alloc]init];
    DCTool * t3 = [[DCTool alloc]init];
    
    //这里配置未改
    [t1 updateWithConfig:^(DCConfig *config) {
    }];
    [t1 sayConfig];
    
    //这里配置已改
    [t2 updateWithConfig:^(DCConfig *config) {
        config.name =@"赵大财改";
        config.age = 1989;
        config.job = [NSString stringWithFormat:@"%@找工作",config.job];
    }];
    [t2 sayConfig];
    
    //这里链式修改配置
    [t3 updateWithConfig:^(DCConfig *config) {
        config.age = 1989;
        config.changeJob(@"奔跑吧 工作").changeName(@"赵大财神");
    }];
    [t3 sayConfig];
    
    
    // Do any additional setup after loading the view, typically from a nib.
}


- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}


@end

File: SwiftPOP/SwiftPOP/ViewController.swift
//
//  ViewController.swift
//  SwiftPOP
//
//  Created by point on 2017/5/14.
//  Copyright © 2017年 dacai. All rights reserved.
//

import UIKit

class ViewController:UIViewController, DCProtocol  , Emitterable{

    override func viewDidLoad() {
        super.viewDidLoad()
        test()
        // Do any additional setup after loading the view, typically from a nib.
        let demoView = DCView.loadFromNib()
        demoView.frame = CGRect(x: 0, y: 0, width: 100, height: 100)
     
        view.addSubview(demoView)

    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }
    
    
    override func touchesBegan(_ touches: Set<UITouch>, with event: UIEvent?) {
        startEmittering(CGPoint(x: view.bounds.height * 0.5 , y: view.bounds.height))
    }


}

